support-frontend/assets/pages/[countryGroupId]/components/checkoutComponent.tsx

```diff
@@ -900,111 +899,18 @@ export function CheckoutComponent({
-										preventDefaultValidityMessage(event.currentTarget);
-										const validityState = event.currentTarget.validity;
-										if (validityState.valid) {
-											setEmailError(undefined);
-										} else {
-											if (validityState.valueMissing) {
-												setEmailError('Please enter your email address.');
-											} else {
-												setEmailError('Please enter a valid email address.');
-											}
-										}
-									}}
-								/>
-							</div>
-							<Signout isSignedIn={isSignedIn} />
-							<>
-								<div>
-									<TextInput
-										id="firstName"
-										data-qm-masking="blocklist"
-										label="First name"
-										value={firstName}
-										autoComplete="given-name"
-										autoCapitalize="words"
-										onChange={(event) => {
-											setFirstName(event.target.value);
-										}}
-										onBlur={(event) => {
-											event.target.checkValidity();
-										}}
-										name="firstName"
-										required
-										maxLength={40}
-										error={firstNameError}
-										pattern={doesNotContainExtendedEmojiOrLeadingSpace}
-										onInvalid={(event) => {
-											preventDefaultValidityMessage(event.currentTarget);
-											const validityState = event.currentTarget.validity;
-											if (validityState.valid) {
-												setFirstNameError(undefined);
-											} else {
-												if (validityState.valueMissing) {
-													setFirstNameError('Please enter your first name.');
-												} else {
-													setFirstNameError('Please enter a valid first name.');
-												}
-											}
-										}}
-									/>
-								</div>
-								<div>
-									<TextInput
-										id="lastName"
-										data-qm-masking="blocklist"
-										label="Last name"
-										value={lastName}
-										autoComplete="family-name"
-										autoCapitalize="words"
-										onChange={(event) => {
-											setLastName(event.target.value);
-										}}
-										onBlur={(event) => {
-											event.target.checkValidity();
-										}}
-										name="lastName"
-										required
-										maxLength={40}
-										error={lastNameError}
-										pattern={doesNotContainExtendedEmojiOrLeadingSpace}
-										onInvalid={(event) => {
-											preventDefaultValidityMessage(event.currentTarget);
-											const validityState = event.currentTarget.validity;
-											if (validityState.valid) {
-												setLastNameError(undefined);
-											} else {
-												if (validityState.valueMissing) {
-													setLastNameError('Please enter your last name.');
-												} else {
-													setLastNameError('Please enter a valid last name.');
-												}
-											}
-										}}
-									/>
-								</div>
-							</>
+
+							<PersonalDetailsFields
+								isEmailAddressReadOnly={isSignedIn}
+								firstName={firstName}
+								setFirstName={(firstName) => setFirstName(firstName)}
```

Comment:
Thats clever over the usual way of passing back an on `onChange` and having to put a handler in the parent. Looks like less code in the parent too. I like that, perhaps get George's opinion too?

Reply:
It's neat here, I thin there are occasions an inline handler function in the setter wouldn't be appropriate, eg. the handler is called from multiple places or it contains a lot of logic, but in this use case it looks good.